lib/grape/middleware/logger/rack_head_override.rb

```diff
@@ -5,7 +5,8 @@ module RackHeadOverride
     def call(env)
       response = super
       status, _, rack = *response
-      response_object = JSON.parse(rack.body.try(:first) || '{}').with_indifferent_access
+      # rescue rack response in case non-json output is given
+      response_object = JSON.parse(rack.body.try(:first) || '{}').with_indifferent_access rescue {}
```

Comment:
Maybe we can wrap the non-json response as value of the error? (not sure what's in there tho...)
```suggestion
      response_object = JSON.parse(rack.body.try(:first) || '{}').with_indifferent_access rescue { error: rack.body.try(:first) }
```